File: backend/src/modules/system/system.controller.ts
import { Body, Controller, Get, Param, Patch, Query, UseGuards } from '@nestjs/common';
import { ApiBearerAuth, ApiOperation, ApiTags } from '@nestjs/swagger';
import { SystemService } from './system.service';
import { JwtAuthGuard } from '@/common/guards/jwt-auth.guard';
import { RbacGuard } from '@/common/guards/rbac.guard';
import { Permissions } from '@/common/decorators/roles.decorator';
import { Type } from 'class-transformer';
import { IsInt, IsOptional, IsString, Min } from 'class-validator';
import { ApiPropertyOptional } from '@nestjs/swagger';

class QueryAuditLogDto {
  @ApiPropertyOptional({ default: 1 })
  @IsOptional()
  @Type(() => Number)
  @IsInt()
  @Min(1)
  page?: number = 1;

  @ApiPropertyOptional({ default: 20 })
  @IsOptional()
  @Type(() => Number)
  @IsInt()
  pageSize?: number = 20;

  @ApiPropertyOptional()
  @IsOptional()
  @IsString()
  module?: string;
}

@ApiTags('系统管理')
@ApiBearerAuth()
@UseGuards(JwtAuthGuard, RbacGuard)
@Controller('system')
export class SystemController {
  constructor(private systemService: SystemService) {}

  @Get('roles')
  @Permissions('system:manage')
  @ApiOperation({ summary: '获取角色列表' })
  getRoles() {
    return this.systemService.getRoles();
  }

  @Get('permissions')
  @Permissions('system:manage')
  @ApiOperation({ summary: '获取权限列表' })
  getPermissions() {
    return this.systemService.getPermissions();
  }

  @Get('audit-logs')
  @Permissions('system:manage')
  @ApiOperation({ summary: '获取操作日志' })
  getAuditLogs(@Query() query: QueryAuditLogDto) {
    return this.systemService.getAuditLogs({
      page: query.page ?? 1,
      pageSize: query.pageSize ?? 20,
      module: query.module,
    });
  }

  @Get('email-templates')
  @Permissions('system:manage')
  @ApiOperation({ summary: '获取邮件模板' })
  getEmailTemplates() {
    return this.systemService.getEmailTemplates();
  }

  @Patch('email-templates/:id')
  @Permissions('system:manage')
  @ApiOperation({ summary: '更新邮件模板' })
  updateEmailTemplate(
    @Param('id') id: string,
    @Body() body: { subject?: string; body?: string; isActive?: boolean },
  ) {
    return this.systemService.updateEmailTemplate(id, body);
  }
}

File: backend/src/main/java/com/company/hr/controller/SystemController.java
package com.company.hr.controller;

import com.company.hr.common.PermissionCheck;
import com.company.hr.common.dto.ApiResponse;
import com.company.hr.common.dto.PagedResponse;
import com.company.hr.entity.AuditLog;
import com.company.hr.entity.EmailTemplate;
import com.company.hr.entity.Permission;
import com.company.hr.entity.Role;
import com.company.hr.service.SystemService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/system")
@RequiredArgsConstructor
@Tag(name = "系统管理")
public class SystemController {

    private final SystemService systemService;

    @GetMapping("/roles")
    @PermissionCheck("system:manage")
    @Operation(summary = "获取所有角色")
    public ApiResponse<List<Role>> getRoles() {
        return ApiResponse.ok(systemService.getRoles());
    }

    @GetMapping("/permissions")
    @PermissionCheck("system:manage")
    @Operation(summary = "获取所有权限")
    public ApiResponse<List<Permission>> getPermissions() {
        return ApiResponse.ok(systemService.getPermissions());
    }

    @GetMapping("/audit-logs")
    @PermissionCheck("system:manage")
    @Operation(summary = "查询审计日志")
    public ApiResponse<PagedResponse<AuditLog>> getAuditLogs(
            @RequestParam(defaultValue = "1") int page,
            @RequestParam(defaultValue = "20") int pageSize,
            @RequestParam(required = false) String module) {
        return ApiResponse.ok(systemService.getAuditLogs(page, pageSize, module));
    }

    @GetMapping("/email-templates")
    @PermissionCheck("system:manage")
    @Operation(summary = "获取邮件模板")
    public ApiResponse<List<EmailTemplate>> getEmailTemplates() {
        return ApiResponse.ok(systemService.getEmailTemplates());
    }

    @PatchMapping("/email-templates/{id}")
    @PermissionCheck("system:manage")
    @Operation(summary = "更新邮件模板")
    public ApiResponse<EmailTemplate> updateEmailTemplate(
            @PathVariable String id,
            @RequestBody Map<String, Object> updates) {
        String subject = updates.containsKey("subject") ? (String) updates.get("subject") : null;
        String body = updates.containsKey("body") ? (String) updates.get("body") : null;
        Boolean isActive = updates.containsKey("isActive") ? (Boolean) updates.get("isActive") : null;
        return ApiResponse.ok(systemService.updateEmailTemplate(id, subject, body, isActive));
    }
}
